core/src/main/java/org/geysermc/geyser/network/GameProtocol.java

```diff
@@ -170,14 +194,105 @@ public static String getAllSupportedJavaVersions() {
+                // Ignored only when serverbound
+                .updateSerializer(MobArmorEquipmentPacket.class, new MobArmorEquipmentSerializer_v291() {
                     @Override
-                    public void deserialize(ByteBuf buffer, BedrockCodecHelper helper, TrimDataPacket packet) {
+                    public void deserialize(ByteBuf buffer, BedrockCodecHelper helper, MobArmorEquipmentPacket packet) {
+                    }
+                })
+                // Valid serverbound packets where reading of some fields can be skipped
+                .updateSerializer(MobEquipmentPacket.class, new MobEquipmentSerializer_v291() {
+                    @Override
+                    public void deserialize(ByteBuf buffer, BedrockCodecHelper helper, MobEquipmentPacket packet) {
+                        packet.setRuntimeEntityId(VarInts.readUnsignedLong(buffer));
+                        fakeItemRead(buffer);
+                        packet.setInventorySlot(buffer.readUnsignedByte());
+                        packet.setHotbarSlot(buffer.readUnsignedByte());
+                        packet.setContainerId(buffer.readByte());
                     }
                 })
                 .build();
     }
 
+    /**
+     * Fake reading an item from the buffer to improve performance.
+     * 
+     * @param buffer
+     */
+    private static void fakeItemRead(ByteBuf buffer) {
+        int id = VarInts.readInt(buffer); // Runtime ID
+        if (id == 0) { // nothing more to read
+            return;
+        }
+        buffer.skipBytes(2); // count
+        VarInts.readUnsignedInt(buffer); // damage
+        boolean hasNetId = buffer.readBoolean();
+        if (hasNetId) {
+            VarInts.readInt(buffer);
+        }
+
+        VarInts.readInt(buffer); // Block runtime ID
+        int streamSize = VarInts.readUnsignedInt(buffer);
+        buffer.skipBytes(streamSize);
+    }
+
+    private static <T extends BedrockPacket> BedrockPacketSerializer<T> setIllegalSerializer() {
+        return new BedrockPacketSerializer<T>() {
+            @Override
+            public void serialize(ByteBuf buffer, BedrockCodecHelper helper, T packet) {
+                throw new IllegalArgumentException("Server tried to send unused packet " + packet.getClass().getSimpleName() + "!");
+            }
+    
+            @Override
+            public void deserialize(ByteBuf buffer, BedrockCodecHelper helper, T packet) {
+                throw new IllegalArgumentException("Client tried to send unused packet " + packet.getClass().getSimpleName() + "!");
+            }
+        };
+    }
+
+    private static <T extends BedrockPacket> BedrockPacketSerializer<T> setIgnoredSerializer() {
+        return new BedrockPacketSerializer<T>() {
+            @Override
+            public void serialize(ByteBuf buffer, BedrockCodecHelper helper, BedrockPacket packet) {
```

Comment:
Would it maybe be worth it to log ignored packets to debug log here?

Reply:
Would get kinda spammy as the client is doing nothing wrong by sending them. We just have no use for them.